src/sagemaker_containers/_transformer.py

```diff
@@ -123,10 +124,20 @@ def __init__(self, model_fn=None, input_fn=None, predict_fn=None, output_fn=None
             input_fn (fn): Takes request data and de-serializes the data into an object for prediction.
             predict_fn (fn): Function responsible for model predictions.
             output_fn (fn): Function responsible to serialize the prediction for the response.
+            transform_fn (fn): Function responsible for taking input data and returning a prediction
+                as a serialized response. This function takes the place of ``input_fn``,
+                ``predict_fn``, and ``output_fn``.
             error_class (Exception): Error class used to separate framework and user errors.
         """
         self._model = None
         self._model_fn = _functions.error_wrapper(model_fn, error_class) if model_fn else default_model_fn
+
+        if transform_fn and (input_fn or predict_fn or output_fn):
+            raise ValueError('Cannot use transform_fn implementation with input_fn, predict_fn, and/or output_fn')
+
+        self._transform_fn = _functions.error_wrapper(transform_fn, error_class) \
+            if transform_fn else self._default_transform_fn
```

Comment:
nit: ternaries should be used only for one liners https://github.com/google/styleguide/blob/gh-pages/pyguide.md#2114-decision